### ISSUE TITLE
Redundant notifications

### ISSUE THREAD

**Issue by dcsjapan** [flarum/core#1091]
This is an issue we've discussed before in Gitter chat, but I thought I should make an issue of it so it doesn't get lost in the shuffle.

I frequently receive two or three notifications (both bell alerts and emails) as the result of a single user post. One or two of the redundant notifications come from **Subscriptions** and **Mentions**, but it seems the notification infrastructure is also implicated, as we can see in this dropdown ...

![screenie1](https://cloud.githubusercontent.com/assets/13129997/21384582/7b57cd4c-c7ae-11e6-86e2-b8a38ac67ba4.png)

... and in my mailbox ...

![screenie2](https://cloud.githubusercontent.com/assets/13129997/21384590/80e27c3a-c7ae-11e6-9894-2cadd6785583.png)

One response by Liberty has resulted in:

- A notification from **Mentions**
- A reply notification
- A notification from **Subscriptions**

For some reason, the emails are listed in the reverse order. That's probably not important, but I guess there's a tiny chance it might be indicative of the cause of this issue.

The notification infrastructure needs a reliable mechanism that will allow it to recognize when it's about to output multiple notifications for a single post, and only send the most important one:

- A reply notification should obviate any other notifications regarding the same post.
- A **Subscriptions** notification should obviate a **Mentions** notification for the same post.

So the order of priority is **replies** > **Subscriptions** > **Mentions**.

**Comment by dcsjapan** [comment]
Follow up info:

That unread notification from BlackSheep in the first screenshot above was actually a reply to a post in a thread I was following ... and yet I received only a **Subscriptions** notification in that case. 😦 

Could it be that redundant notifications only happen when Flarum has to deal with notifications from all three sources? I'll keep an eye out for counterexamples and update this thread if I notice anything that contradicts this theory.

**Comment by dcsjapan** [comment]
Related to https://github.com/flarum/core/issues/868.

**Comment by franzliedke** [comment]
Is this a dupe of #439?

**Comment by dcsjapan** [comment]
@franzliedke Yes, I suppose it is. 😛 I thought I had filed this already, but I couldn't find it.

I'll close that older one, since this one is much more detailed. Plus, screenshots! :grip

**Comment by dcsjapan** [comment]
Sorry about that ... I tried to click on an emoji dropdown and ended up closing the issue instead. 😬
